src/containerapp/azext_containerapp/containerapp_maintenance_config_decorator.py

```diff
@@ -0,0 +1,114 @@
+        return self.get_param('resource_group_name')
+
+    def get_argument_weekday(self):
+        return self.get_param('weekday')
+
+    def get_argument_start_hour_utc(self):
+        return self.get_param('start_hour_utc')
+
+    def get_argument_duration(self):
+        return self.get_param('duration')
+
+
+class MaintenanceConfigPreviewDecorator(MaintenanceConfigDecorator):
+    def validate_arguments(self):
+        if self.get_argument_start_hour_utc() is not None:
+            if not (0 <= int(self.get_argument_start_hour_utc()) <= 23):
+                raise ValidationError("Start hour must be an integer from 0 to 23")
+
+        if self.get_argument_duration() is not None:
+            if not (8 <= int(self.get_argument_duration()) <= 24):
+                raise ValidationError("Duration must be an integer from 8 to 24")
+
+        if self.get_argument_weekday is not None:
+            if self.get_argument_weekday().lower() not in ["monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday"]:
+                raise ValidationError("Weekday must be a day of the week")
+
+    def construct_payload(self, forUpdate=False):
+        if forUpdate:
+            self.existing_maintenance_config_def = self.client.show(
+                cmd=self.cmd,
+                resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name())
+
+            self.maintenance_config_def = deepcopy(self.existing_maintenance_config_def)
+
+        if self.get_argument_start_hour_utc() is not None:
+            self.maintenance_config_def["properties"]["scheduledEntries"][0]["startHourUtc"] = self.get_argument_start_hour_utc()
+        if self.get_argument_duration() is not None:
+            self.maintenance_config_def["properties"]["scheduledEntries"][0]["durationHours"] = self.get_argument_duration()
+        if self.get_argument_weekday() is not None:
+            self.maintenance_config_def["properties"]["scheduledEntries"][0]["weekDay"] = self.get_argument_weekday()
+
+    def add(self):
+        try:
+            return self.client.add(
+                cmd=self.cmd,
+                resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name(),
+                maintenance_config_envelope=self.maintenance_config_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def update(self):
+        try:
+            return self.client.update(
+                cmd=self.cmd,
+                resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name(),
+                maintenance_config_envelope=self.maintenance_config_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
```

Comment:
What is the difference betweent self.client.add and self.client.update? Are they using `Patch` method both?

Reply:
combined into create_or_update